wasm-glow-ballers-lib/src/lib.rs

```diff
@@ -5,13 +5,109 @@ use std::collections::HashMap;
 use error::Error;
 use walrus::Module;
 
-pub fn mapping(wasm: &[u8]) -> Result<HashMap<String, usize>, Error> {
+pub enum MappingItemType {
+    Global,
+    Function,
+    Memory,
+    Table,
+}
+
+// // Creates a mapping of all exported and imported globals (original use case of the library)
+// pub fn mapping(wasm: &[u8]) -> Result<HashMap<String, usize>, Error> {
+//     mapping_with_type(wasm, MappingItemType::Global)
+// }
+
+// Creates a mapping of all exported and imported items of the specified type
+pub fn mapping_with_type(
+    wasm: &[u8],
+    item_type: MappingItemType,
```

Comment:
This function takes ownership of `item_type`, while eg. `mapping_exports_with_type_and_module` does not. Aim for more consistency?